executor/aggfuncs/builder.go

```diff
@@ -50,6 +50,24 @@ func Build(aggFuncDesc *aggregation.AggFuncDesc, ordinal int) AggFunc {
 
 // buildCount builds the AggFunc implementation for function "COUNT".
 func buildCount(aggFuncDesc *aggregation.AggFuncDesc, ordinal int) AggFunc {
+	base := baseAggFunc{
+		args:    aggFuncDesc.Args,
+		ordinal: ordinal,
+	}
+	switch aggFuncDesc.Mode {
+	case aggregation.DedupMode:
+		return nil // not implemented yet.
+
+		// Build count functions which consume the original data and update their
+		// partial results.
+	case aggregation.CompleteMode, aggregation.Partial1Mode:
+		return &countOriginal{baseCount{base}}
```

Comment:
This implementation has not properly handled the `distinct` property. We should implement other `AggFunc`s for `COUNT` to count the distinct and non-NULL values in a data group.

Reply:
Something like `countOriginalWithDistinct` ?

Reply:
yes

Reply:
I think I should port the `distinctChecker` to the new aggregate framework which depends on your pr https://github.com/pingcap/tidb/pull/7006 .

Reply:
@Xuanwo Maybe what you need is `HashChunkRow()` defined in **util/codec/codec.go**?

Reply:
Looks cool, let me check that.